### PR TITLE
Initial commit of git extension

### PR DESCRIPTION
**Pull request by cunningt**
I noticed there wasn't a git extension and I wanted to get involved so I started with git.

I think I covered all of the steps in the Camel Quarkus Installation guide, but I haven't been able to run integration-tests/git with mvn -Pnative verify - all of the integration tests are failing for me in native profile.

### CODE REVIEW

**Comment by ppalaga** [comment]
@cunningt Quarkus has a jgit extension https://github.com/quarkusio/quarkus/tree/master/extensions/jgit that probably solves the issues you see. Try to add `io.quarkus:quarkus-jgit` dependency to your runtime modules and `io.quarkus:quarkus-jgit-deployment` to your deployment module.

File: integration-tests/git/src/main/java/org/apache/camel/quarkus/component/git/it/GitResource.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response post(String message) throws Exception {
+        LOG.infof("Sending to git: %s", message);
+        Exchange initResponse = producerTemplate.request("git:target/" + repoName + "?operation=init",
+                exchange -> exchange.getIn());
+
+        // We're gonna put a real file in here
+        File newFile = new File("target/" + repoName + "/" + fileName);
+        BufferedWriter writer = new BufferedWriter(new FileWriter(newFile));
```
Comment (ppalaga) [inline integration-tests/git/src/main/java/org/apache/camel/quarkus/component/git/it/GitResource.java]: Please enclose the writer in a try() statement

**Comment by davsclaus** [comment]
ok to test

**Comment by ppalaga** [comment]
@cunningt could you please rebase?

**Comment by ppalaga** [comment]
Superceded by https://github.com/apache/camel-quarkus/pull/1548

**Pull request closed without merging**